8.cpp

```diff
@@ -4,11 +4,11 @@
 using namespace std;
 class Solution {
  public:
-  int myAtoi(string str) {  //��һ��ʧ���ˣ���������long long���Ų���
+  int myAtoi(string str) {  //第一遍失败了，最后结果可能long long都放不下
     int positive_num = 0;
     long long res = 0;
     auto i = str.begin();
-    for (; i != str.end(); i++)  //��ͷ�հ�ȥ��
+    for (; i != str.end(); i++)  //开头空白去掉
```

Comment:
I suggest rewrite this block as:
```C++
for (; iter != str.end() && *iter == ' '; ++iter)
  ;
if (iter == str.end()) return 0; 
// decide the sign
if (*iter == '+') ++iter;
else if (*iter == '-') {
  is_positive = false;
  ++iter;
}
...other part...
```

Reply:
Or
```C++
auto iter = std::find_if(str.begin(), str.end(), [](char c) { return c != ' '; });
```

Reply:
BTW, use ```++iter``` instead of ```iter++``` when possible. The former is potentially more efficient. (Do you remember the distinction?)

Reply:
done